File: app/core/processing_status.py
from typing import Literal

AssessmentProcessingStatus = Literal["queued", "processing", "completed", "failed"]

QUEUED: AssessmentProcessingStatus = "queued"
PROCESSING: AssessmentProcessingStatus = "processing"
COMPLETED: AssessmentProcessingStatus = "completed"
FAILED: AssessmentProcessingStatus = "failed"


def is_ready(status: str) -> bool:
    return status == COMPLETED

File: frontend/src/lib/processingStatus.ts
import type { AssessmentProcessingStatus } from "../api/types"

export function isProcessingStatus(status: AssessmentProcessingStatus): boolean {
  return status === "queued" || status === "processing"
}

export function processingStatusLabel(status: AssessmentProcessingStatus): string {
  switch (status) {
    case "queued":
      return "En cola"
    case "processing":
      return "Analizando"
    case "failed":
      return "Error"
    default:
      return "Listo"
  }
}
